src/Microsoft.AspNetCore.Routing/Template/TemplateMatcher.cs

```diff
@@ -82,20 +81,25 @@ public bool TryMatch(PathString path, RouteValueDictionary values)
             //
             // On the second pass, we're almost certainly going to match the URL, so go ahead and allocate the 'values'
             // and start capturing strings. 
-            foreach (var requestSegment in pathTokenizer)
+            foreach (var pathSegment in pathTokenizer)
```

Comment:
We usually refer to path vs route across the routing code base, requestSegment and routeSegment are actually pretty similar and it was causing me trouble to follow through the code, so I went and made it clearer